diff --git a/lib/index.js b/lib/index.js
index eb116c240965cd342adef9f18e8cf1709ef5ec16..c615db78a42a1d6e9f6e3141b1654dc1a348bba2 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -69,6 +69,11 @@ async function downloadHeaders(version, distDir) {
   const extractedDir = path.join(distDir, 'node_headers');
   if (fs.existsSync(extractedDir)) {
     fs.renameSync(extractedDir, headersPath);
+  } else if (fs.existsSync(path.join(distDir, 'include'))) {
+    if (!fs.existsSync(headersPath)) {
+      fs.mkdirSync(headersPath, { recursive: true });
+    }
+    fs.renameSync(path.join(distDir, 'include'), path.join(headersPath, 'include'));
   }
 
   fs.unlinkSync(zipPath);
